src/Test.Plugins/ContextGeneratingHelper.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.using System;
+
+using Microsoft.CodeAnalysis.Sarif;
+using Microsoft.CodeAnalysis.Sarif.WorkItems;
+using Microsoft.CodeAnalysis.Test.Plugins;
+using Xunit;
+
+namespace Test.Plugins
+{
+    public class ContextGeneratingHelper
+    {
+        [Fact]
+        public void GenerateSampleContextToTestPlugin()
+        {
+            var sarifWorkItemContext = new SarifWorkItemContext();
+            sarifWorkItemContext.ProjectUri = new System.Uri("https://github.com/michaelcfanning/bug-dummy");
+            sarifWorkItemContext.SecurityToken = "xxx";
+            sarifWorkItemContext.AddWorkItemModelTransformer(new TestWorkItemModelTransformer());
+            sarifWorkItemContext.SetProperty(TestWorkItemModelTransformer.AdditionalTags, new StringSet(new[] { "extensionAddedTestTag" }));
+            sarifWorkItemContext.SaveToXml(@"e:\repros\AddTagForGithub.xml");
```

Comment:
>e:\repros\AddTagForGithub.xml [](start = 45, length = 29)

`SarifWorkItemContext` derives from `PropertiesDictionary`, whose `SaveToXml` method really will try to write to this file. It won't work on machines with no E: drive or (even if there is an E: drive) on a test machine where the account running the tests doesn't have permission to access this path. We need to be able to inject a mock file system into `PropertiesDictionary`.

Reply:
yep, this test is guaranteed to fail on my machine since I don't have a local e drive.

---
In reply to: [381027491](https://github.com/microsoft/sarif-sdk/pull/1775#discussion_r381027491) [](ancestors = 381027491)